lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/authentication/KerberosAuthentication.java

```diff
@@ -58,8 +58,11 @@ public Subject getSubject()
     public void attemptLogin(Subject subject)
     {
         try {
-            LoginContext loginContext = new LoginContext("", subject, null, configuration);
-            loginContext.login();
+            synchronized (subject.getPrivateCredentials()) {
```

Comment:
Why on `subject.getPrivateCredentials()` and not on `subject`?
Why do we need synchronize at all?

please capture reasoning as a code comment